streaming/src/main/scala/org/apache/spark/streaming/receiver/Receiver.scala

```diff
@@ -83,7 +84,12 @@ import org.apache.spark.storage.StorageLevel
  * }}}
  */
 @DeveloperApi
-abstract class Receiver[T](val storageLevel: StorageLevel) extends Serializable {
+abstract class Receiver[T](val storageLevel: StorageLevel)(implicit t: ClassTag[T])
```

Comment:
As far as I know, this change will indeed broke **custom receiver** implementation in **Java**, just like the update in **JavaCustomReceiver.java**. Besides, in my humble opinion, I did not encounter any problem in my local test.